app/models/case_contact.rb

```diff
@@ -145,6 +145,12 @@ def active_or_expenses?
 
   scope :no_drafts, ->(checked) { (checked == 1) ? where(status: "active") : all }
 
+  scope :drafts_for_removal, -> {
+    where.not(status: "active")
+      .where("case_contacts.created_at < ?", 1.week.ago)
+      .or(where(draft_case_ids: []).where("case_contacts.created_at < ?", 1.day.ago))
+  }
+
```

Comment:
We aren't the most consistent with it in this code base but I think we should keep business logic out of our scopes. In this case the designation of what is "for removal" is that business logic. 

Most of the other scopes are pretty generic.

```rb
scope :created_max_ago, ->(time_range = nil) {
  where("case_contacts.created_at > ?", time_range) if time_range.present?
}
```

I think this query belongs in a service object or maybe just some plain ruby object.

Reply:
Got it. I'll move it into a service class and do the destroy there as well so the rake task just calls the service method. I'll update tests accordingly